src/edu/csus/ecs/pc2/ui/EditProblemSandboxPane.java

```diff
@@ -192,6 +182,40 @@ protected void initializeFields(Problem inProblem) {
         }
     }
     
+    /**
+     * Populate controls with problem's sandbox information
+     * 
+     * @param problem - the problem to use to populate the controls.  If null, uses defaults
+     */
+    private void populateGUI(Problem problem) {
+        
+        // If adding a problem, we have to determine a default sandbox type
+        if(problem == null) {
+            // If sandboxes are desired and supported, we make the internal sandbox the default
+            if(Problem.DEFAULT_SANDBOX_TYPE == SandboxType.PC2_INTERNAL_SANDBOX && isCheckSandboxSupport()) {
+                getUsePC2SandboxRadioButton().setSelected(true);
+            } else {
+                getUseNoSandboxRadioButton().setSelected(true);
+            }
+            getPC2SandboxOptionMemLimitTextbox().setText(Integer.toString(Problem.DEFAULT_MEMORY_LIMIT_MB));
+       } else {
+            //activate the appropriate sandbox selection button (the buttons are in a group so selecting one will deselect the others)
+            if (problem.getSandboxType() == SandboxType.NONE) {
+                getUseNoSandboxRadioButton().setSelected(true);
+                
+            } else if (problem.getSandboxType() == SandboxType.PC2_INTERNAL_SANDBOX) {
+                isCheckSandboxSupport();
```

Comment:
It bothers me greatly to see a call to a method (`isCheckSandboxSupport()`) which returns a boolean which is then ignored. Shouldn't a return value of false here cause some kind of notification to the user?

Reply:
The message to the user is issued in the isCheckSandboxSupport() method if sandboxes are not supported.  If sandboxes are NOT supported, we do not want to change an existing problem's setting, we just want to notify the user that an existing problem uses sandboxes, and, they're not supported on the current system.  So, there's really nothing to be done in this case if they're not supported.  I don't think it's bad programming style to not use a return value.  Maybe change the name of the method to: isCheckSandboxSupportAndNotify()?

Reply:
I don't think changing the name of the method is necessary (or appropriate).  I think boolean accessors which query a status ("isSandboxSupported" in this case) shouldn't have side effects -- they should return a yes/no answer to the question implied by the method name.  The _calling_ code should then do what is necessary/appropriate depending on the answer (such as notifying the user of an issue).   

I will concede that this is somewhat of a personal prejudice; I spent a lot of time teaching students about "rules of good programming" and one of those "rules" is "functions shouldn't have side-effects".  One of the primary reasons for this is that it greatly reduces issues with formal verification.

Having said that, I also concede that the idea that we will _ever_ get to the point of applying any kind of formal verification tools/techniques to PC2 is, at best, laughable -- so this really isn't a huge issue and I don't think any change is needed (hence the reason I went ahead and "Approved" the PR).  I just couldn't help myself from pointing it out as a general design principle (although I grant, not one that everyone agrees on...)

Reply:
Understood.  So does this mean that you do approve the PR.  If so, then we can merge it, correct?